packages/server/src/api/routes/tests/screen.spec.ts

```diff
@@ -205,6 +206,120 @@ describe("/screens", () => {
       })
     })
+
+    describe("workspace apps", () => {
+      let featureCleanup: () => void
+      beforeAll(() => {
+        featureCleanup = features.testutils.setFeatureFlags("*", {
+          WORKSPACE_APPS: true,
+        })
+      })
+
+      afterAll(() => {
+        featureCleanup()
+      })
+
+      it("should not allow deleting the last screen", async () => {
+        const { workspaceApp } = await config.api.workspaceApp.create(
+          structures.workspaceApps.workspaceApp()
+        )
+        const screen = await config.api.screen.save({
+          ...basicScreen(),
+          workspaceAppId: workspaceApp._id,
+        })
+
+        await config.api.screen.destroy(screen._id!, screen._rev!, {
+          status: 409,
+          body: { message: "Cannot delete the last screen in a workspace app" },
+        })
+      })
+
+      it("should allow deleting other screens", async () => {
+        const { workspaceApp } = await config.api.workspaceApp.create(
+          structures.workspaceApps.workspaceApp()
+        )
+        const screens = await Promise.all(
+          Array.from({ length: 3 }).map(() =>
+            config.api.screen.save({
+              ...basicScreen(),
+              workspaceAppId: workspaceApp._id,
+            })
+          )
+        )
+
+        function popRandomScreen() {
+          const index = Math.floor(Math.random() * screens.length)
+          return screens.splice(index, 1)[0]
+        }
+
+        let screenToDelete = popRandomScreen()
+        await config.api.screen.destroy(
+          screenToDelete._id!,
+          screenToDelete._rev!
+        )
+        screenToDelete = popRandomScreen()
+        await config.api.screen.destroy(
+          screenToDelete._id!,
+          screenToDelete._rev!
+        )
+        screenToDelete = popRandomScreen()
+        await config.api.screen.destroy(
+          screenToDelete._id!,
+          screenToDelete._rev!,
+          { status: 409 }
+        )
+      })
+
+      it("should allow deleting screens for other apps", async () => {
+        const { workspaceApp: workspaceApp1 } =
+          await config.api.workspaceApp.create(
+            structures.workspaceApps.workspaceApp()
+          )
+        const { workspaceApp: workspaceApp2 } =
+          await config.api.workspaceApp.create(
+            structures.workspaceApps.workspaceApp()
+          )
+        const app1Screens = await Promise.all(
+          Array.from({ length: 3 }).map(() =>
+            config.api.screen.save({
+              ...basicScreen(),
+              workspaceAppId: workspaceApp1._id,
+            })
+          )
+        )
+        // Other screens
+        await Promise.all(
+          Array.from({ length: 2 }).map(() =>
+            config.api.screen.save({
+              ...basicScreen(),
+              workspaceAppId: workspaceApp2._id,
+            })
+          )
+        )
+
+        function popRandomScreen(screens: Screen[]) {
```

Comment:
Probably worth extracting this to the top level but I don't feel super strongly about it.